public/src/MockGRPCManager.ts

```diff
@@ -456,6 +456,14 @@ export class MockGrpcManager {
         return this.grpcSend<Submissions>(submissions)
     }
 
+    public getSubmission(courseID: number, submissionID: number): Promise<IGrpcResponse<Submission>> {
+        if (!this.enrollments.getEnrollmentsList().find(e => e.getCourseid() === courseID && e.getUserid() === this.currentUser?.getId())) {
+            return this.grpcSend<Submission>(null, new Status().setCode(StatusCode.NOT_FOUND))
```

Comment:
This looks a bit odd to do a send with `null`. Could we instead make a local function that simply returns the expected type necessary for the caller to handle a not found reply.